ProcessName not shown in analysis

**Describe the bug**
ProcessName is not shown in Ports findings (ASA > Results > Ports).

**To Reproduce**
Steps to reproduce the behavior:
1. Open asa gui with admin rights
2. Scan the computer with asa (Select only Ports collector)
3. Open PowerShell window and open a new port:
    >$Listener = [System.Net.Sockets.TcpListener]9999;
    >$Listener.Start();
4. Check with netstat -abn that port 9999 is opened and see what process is listening this port (pwsh.exe)
5. Scan the computer again with asa (Select only Ports collector)
6. Open Results and compare these two baselines and select Ports
4. Port 9999 is shown but the ProcessName is empty

**Expected behavior**
ProcessName should be shown as it is available in netstat -abn. It is kind of essential information, which would help a lot in the analysis phase.

**Screenshots**
![image](https://user-images.githubusercontent.com/74306109/116969563-28501500-acbf-11eb-8e24-6ec4bd619a75.png)


**System Configuration (please complete the following information):**
 - OS: Windows
 - OS Version: Windows 10 20H2 
 - Application Version: ASA 2.2.78
 - CLI or GUI: gui

**Additional Context**
-

I tried to test this:
```
C:\Users\masty> dotnet tool uninstall -g Microsoft.CST.AttackSurfaceAnalyzer.CLI
C:\Users\masty> dotnet tool install -g --version 2.3.142-beta Microsoft.CST.AttackSurfaceAnalyzer.CLI
warning NU1603: restore depends on Microsoft.CST.AttackSurfaceAnalyzer.CLI (>= 2.3.142-beta) but Microsoft.CST.AttackSurfaceAnalyzer.CLI 2.3.142-beta was not found. An approximate best match of Microsoft.CST.AttackSurfaceAnalyzer.CLI 2.3.142-beta-ge7cc357c64 was resolved.
You can invoke the tool using the following command: asa
Tool 'microsoft.cst.attacksurfaceanalyzer.cli' (version '2.3.142-beta-ge7cc357c64') was successfully installed.

```
But when I ran "asa gui" I got:
```
C:\Users\masty> asa gui
[09:52:49 INF] AttackSurfaceAnalyzer v.2.3.142-beta+e7cc357c64
[09:52:49 FTL] Database has schema version 10 but database has schema version 12.
[09:52:49 FTL] CouldNotSetupDatabase 
C:\Users\masty>

```

--- a/Lib/Collectors/OpenPortCollector.cs
+++ b/Lib/Collectors/OpenPortCollector.cs
@@ -23,21 +23,6 @@ namespace Microsoft.CST.AttackSurfaceAnalyzer.Collectors
 
         public override bool CanRunOnPlatform()
         {
-            try
-            {
-                var osRelease = File.ReadAllText("/proc/sys/kernel/osrelease") ?? "";
-                osRelease = osRelease.ToUpperInvariant();
-                if (osRelease.Contains("MICROSOFT") || osRelease.Contains("WSL"))
-                {
-                    Log.Error("OpenPortCollector cannot run on WSL until https://github.com/Microsoft/WSL/issues/2249 is fixed.");
-                    return false;
-                }
-            }
-            catch (Exception)
-            {
-                /* OK to ignore, expecting this on non-Linux platforms. */
-            };
-
             return RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
         }
 
@@ -65,7 +50,7 @@ namespace Microsoft.CST.AttackSurfaceAnalyzer.Collectors
         {
             try
             {
-                var result = ExternalCommandRunner.RunExternalCommand("ss", "-ln");
+                var result = ExternalCommandRunner.RunExternalCommand("ss", "-lnp");
 
                 foreach (var _line in result.Split('\n'))
                 {
@@ -92,11 +77,31 @@ namespace Microsoft.CST.AttackSurfaceAnalyzer.Collectors
                         {
                             var transport = parts[0].ToUpperInvariant().Equals("TCP") ? TRANSPORT.TCP : TRANSPORT.UDP;
                             var family = address.Contains('.') ? ADDRESS_FAMILY.InterNetwork : address.Contains(':') ? ADDRESS_FAMILY.InterNetworkV6 : ADDRESS_FAMILY.Unknown;
-                            var obj = new OpenPortObject(portInt, transport, family)
+                            var process = string.Empty;
+
+                            if (parts.Length > 6)
                             {
-                                Address = address
-                            };
-                            HandleChange(obj);
+                                var processNameMatches = Regex.Matches(parts[6], @"""(.*?)"",([0-9]*)");
+                                foreach(Match match in processNameMatches)
+                                {
+                                    int? pid = int.TryParse(match.Groups[2].Value, out int thePid) ? thePid : null;
+                                    var obj = new OpenPortObject(portInt, transport, family)
+                                    {
+                                        Address = address,
+                                        ProcessName = match.Groups[1].Value,
+                                        ProcessId = pid
+                                    };
+                                    HandleChange(obj);
+                                }
+                            }
+                            else
+                            {
+                                var obj = new OpenPortObject(portInt, transport, family)
+                                {
+                                    Address = address
+                                };
+                                HandleChange(obj);
+                            }
                         }
                     }
                 }
@@ -190,10 +195,9 @@ namespace Microsoft.CST.AttackSurfaceAnalyzer.Collectors
                 var obj = new OpenPortObject(endpoint.Port, TRANSPORT.TCP, (ADDRESS_FAMILY)endpoint.AddressFamily)
                 {
                     Address = endpoint.Address.ToString(),
+                    ProcessName = Win32ProcessPorts.ProcessPortMap.Find(x => x.PortNumber == endpoint.Port)?.ProcessName,
+                    ProcessId = Win32ProcessPorts.ProcessPortMap.Find(x => x.PortNumber == endpoint.Port)?.ProcessId
                 };
-
-                obj.ProcessName = Win32ProcessPorts.ProcessPortMap.Find(x => x.PortNumber == endpoint.Port)?.ProcessName;
-
                 HandleChange(obj);
             }
 
@@ -201,10 +205,10 @@ namespace Microsoft.CST.AttackSurfaceAnalyzer.Collectors
             {
                 var obj = new OpenPortObject(endpoint.Port, TRANSPORT.UDP, (ADDRESS_FAMILY)endpoint.AddressFamily)
                 {
-                    Address = endpoint.Address.ToString()
+                    Address = endpoint.Address.ToString(),
+                    ProcessName = Win32ProcessPorts.ProcessPortMap.Find(x => x.PortNumber == endpoint.Port)?.ProcessName,
+                    ProcessId = Win32ProcessPorts.ProcessPortMap.Find(x => x.PortNumber == endpoint.Port)?.ProcessId
                 };
-
-                obj.ProcessName = Win32ProcessPorts.ProcessPortMap.Find(x => x.PortNumber == endpoint.Port)?.ProcessName;
 
                 HandleChange(obj);
             }

--- a/Lib/Objects/OpenPortObject.cs
+++ b/Lib/Objects/OpenPortObject.cs
@@ -41,7 +41,15 @@ namespace Microsoft.CST.AttackSurfaceAnalyzer.Objects
         /// </summary>
         public int Port { get; set; }
 
+        /// <summary>
+        /// The associated process if known
+        /// </summary>
         public string? ProcessName { get; set; }
+
+        /// <summary>
+        /// The associated process ID if known
+        /// </summary>
+        public int? ProcessId { get; set; }
 
         /// <summary>
         ///     TCP or UDP

